libqpdf/QPDF.cc

```diff
@@ -2151,8 +2151,8 @@ QPDF::resolveObjectsInStream(int obj_stream_number)
 	}
 
 	int num = QUtil::string_to_int(tnum.getValue().c_str());
-	int offset = QUtil::string_to_int(toffset.getValue().c_str());
-	offsets[num] = offset + first;
+	long long offset = QUtil::string_to_int(toffset.getValue().c_str());
```

Comment:
QUtil::string_to_int ireturns int. If assigning to long long, use the version that returns long long.

Reply:
Assuming the current logic is correct (that toffset represents an "int" range), then I think using `string_to_int` to correct and safer (so the 32-bit range checks can take place).  That way the later addition is guaranteed not to overflow (at least on systems where `long long` is wider than `int`, which seems like a safe assumption).

To make this extra clear I can add a comment and/or a `static_cast`?